source/common/network/utility.cc

```diff
@@ -128,18 +129,25 @@ uint32_t Utility::portFromUdpUrl(const std::string& url) {
 
 Address::InstanceConstSharedPtr Utility::parseInternetAddressNoThrow(const std::string& ip_address,
                                                                      uint16_t port, bool v6only) {
+  Address::InstanceConstSharedPtr address;
   sockaddr_in sa4;
   if (inet_pton(AF_INET, ip_address.c_str(), &sa4.sin_addr) == 1) {
     sa4.sin_family = AF_INET;
     sa4.sin_port = htons(port);
-    return std::make_shared<Address::Ipv4Instance>(&sa4);
+    StatusOr<Address::InstanceConstSharedPtr> error_or_address =
+        Address::InstanceFactory::createInstancePtr<Address::Ipv4Instance>(&sa4);
```

Comment:
You can make a helper
```
template<typename SharedT> SharedT makeOrDie(StatusOr<SharedT> obj) {
  RELEASE_ASSERT(obj.ok(), obj.message());
  return *obj;
}
```
and using that:
```
return makeOrDie(Address::InstanceFactory::createInstancePtr<Address::Ipv4Instance>(&sa4));
```
here and likely other similar places.

Reply:
Makes sense.